macaroonbakery/tests/test_utils.py

```diff
@@ -26,6 +26,13 @@ def test_cookie_expires_with_timezone(self):
             ValueError, cookie, 'http://example.com', 'test', 'value',
             expires=timestamp)
 
+    def test_cookie_with_hostname_not_fqdn(self):
+        c = cookie('http://myhost', 'test', 'value')
+        self.assertEqual(c.domain, 'myhost.local')
+
+    def test_cookie_with_hostname_ipc4(self):
+        c = cookie('http://1.2.3.4', 'test', 'value')
+        self.assertEqual(c.domain, '1.2.3.4')
```

Comment:
Let's add a test for an IPv6 address too please, and something that *almost* looks like an IP address (e.g. 1.2.3.4e)